[AppleWin] Remove Windows code from PixelDumpSupportCG

brentfulgham opened the pull request:
#### 9c0ec4ce794727ac17dac792da8086fa1a438643
<pre>
[AppleWin] Remove Windows code from PixelDumpSupportCG
<a href="https://bugs.webkit.org/show_bug.cgi?id=252197">https://bugs.webkit.org/show_bug.cgi?id=252197</a>

Reviewed by Don Olmstead and Darin Adler.

The testing code in PixelDumpSupportCG.h/PixelDumpSupportCG.cpp used to build on
Windows. We don&apos;t need this support anymore, and should remove it.

This change also switches from a raw buffer to a unique_ptr to clean up the
code a bit.

* Tools/DumpRenderTree/cg/PixelDumpSupportCG.cpp:
(computeSHA1HashStringForBitmapContext):
(createBitmapContext):
* Tools/DumpRenderTree/cg/PixelDumpSupportCG.h:
(BitmapContext::createByAdoptingBitmapAndContext):
(BitmapContext::BitmapContext):
(BitmapContext::~BitmapContext): Deleted.
* Tools/DumpRenderTree/ios/PixelDumpSupportIOS.mm:
(createBitmapContextFromWebView): Remove unneeded buffer argument.
* Tools/DumpRenderTree/mac/PixelDumpSupportMac.mm:
(createBitmapContextFromWebView): Ditto.
(createPagedBitmapContext): Ditto.

Canonical link: <a href="https://commits.webkit.org/260265@main">https://commits.webkit.org/260265@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f7ac18a5cc733022edb147ad8273a302a838c475

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/107624 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/16677 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/40519 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/116773 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/116179 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/111518 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/18097 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/7991 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/99795 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/113379 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/13666 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/96859 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/41328 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/95578 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/28486 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/83171 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/9664 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/29838 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/10339 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/6733 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/15818 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/49420 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7096 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/11899 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Tools/DumpRenderTree/cg/PixelDumpSupportCG.h:
```diff
@@ -33,17 +33,8 @@
 typedef struct CGContext* CGContextRef;
-
-#if PLATFORM(COCOA)
 typedef void* PlatformBitmapBuffer;
```
We don’t need this typedef any more. It existed so we could handle the differences between Cocoa and Windows.

brentfulgham replied:
Fixed.

darinadler commented on Tools/DumpRenderTree/cg/PixelDumpSupportCG.h:
```diff
@@ -55,11 +46,7 @@ class BitmapContext : public RefCounted<BitmapContext> {
     {
         if (m_buffer)
-#if PLATFORM(COCOA)
             free(m_buffer);
```
Should probably change this to use `std::unique_ptr` to clean things up some day.

brentfulgham replied:
I took a stab at doing so. I think the implementation is correct, but would appreciate your thoughts. I'll also wait for all platforms to run tests.

Commit:
[AppleWin] Remove Windows code from PixelDumpSupportCG
https://bugs.webkit.org/show_bug.cgi?id=252197

Reviewed by Don Olmstead and Darin Adler.

The testing code in PixelDumpSupportCG.h/PixelDumpSupportCG.cpp used to build on
Windows. We don't need this support anymore, and should remove it.

This change also switches from a raw buffer to a unique_ptr to clean up the
code a bit.

* Tools/DumpRenderTree/cg/PixelDumpSupportCG.cpp:
(computeSHA1HashStringForBitmapContext):
(createBitmapContext):
* Tools/DumpRenderTree/cg/PixelDumpSupportCG.h:
(BitmapContext::createByAdoptingBitmapAndContext):
(BitmapContext::BitmapContext):
(BitmapContext::~BitmapContext): Deleted.
* Tools/DumpRenderTree/ios/PixelDumpSupportIOS.mm:
(createBitmapContextFromWebView): Remove unneeded buffer argument.
* Tools/DumpRenderTree/mac/PixelDumpSupportMac.mm:
(createBitmapContextFromWebView): Ditto.
(createPagedBitmapContext): Ditto.

Canonical link: https://commits.webkit.org/260265@main
